### PR TITLE
coherent: allocate memory in coherent_init()

### PR DESCRIPTION
**Pull request by lyakh**
Allocating objects, intended for use with the coherent API must use specific flags and align allocation size on a cache-line size boundary. To rid users of these details integrate allocation and freeing into coherent_init*() and coherent_free*() respectively.

This was proposed while reviewing #6730

### CODE REVIEW

**Review by lgirdwood — APPROVED** [review]

File: src/include/sof/coherent.h [hunk 1]
```diff
@@ -367,17 +392,28 @@ static inline void coherent_release_thread(struct coherent __sparse_cache *c,
 	}
 }
 
-static inline void __coherent_init_thread(struct coherent *c, const size_t size)
+static inline void *__coherent_init_thread(size_t offset, const size_t size)
 {
+	void *object = rzalloc(SOF_MEM_ZONE_RUNTIME_SHARED, 0, SOF_MEM_CAPS_RAM,
+			       ALIGN_UP(size, PLATFORM_DCACHE_ALIGN));
```
Comment (kv2019i) [inline src/include/sof/coherent.h:398]: A comment would be nice to explain why the alignment is needed.

**Review by kv2019i — APPROVED** [review]
One comment on explaining the alignment, but as the explanation is in the git commit, so maybe that's enough.

**Comment by kv2019i** [comment]
@lyakh  The PR now has conflicts in mixin_mixout.c

File: src/audio/mixin_mixout/mixin_mixout.c [hunk 2]
```diff
@@ -196,19 +196,16 @@ static struct comp_dev *mixout_new(const struct comp_driver *drv,
 		return NULL;
 	}
 
-	memcpy_s(&md->base_cfg, sizeof(md->base_cfg), spec, sizeof(md->base_cfg));
-
 	/* align the allocation size to a cache line for the coherent API */
-	md->mixed_data_info = rzalloc(SOF_MEM_ZONE_RUNTIME_SHARED, 0, SOF_MEM_CAPS_RAM,
-				      ALIGN_UP(sizeof(struct mixed_data_info),
-					       PLATFORM_DCACHE_ALIGN));
+	md->mixed_data_info = coherent_init_thread(struct mixed_data_info, c);
```
Comment (ranj063) [inline src/audio/mixin_mixout/mixin_mixout.c:200]: @lyakh this parts has moved from the mixin_mixout to the module_adapter now

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**